### PR TITLE
The extends option supports an extended constructor

### PR DESCRIPTION
**Pull request by pespantelis**
#### Related to #2701

### CODE REVIEW

File: test/unit/specs/util/options_spec.js [hunk 1]
```diff
@@ -282,7 +282,8 @@ describe('Util - Option merging', function () {
     var f1 = function () {}
     var f2 = function () {}
     var f3 = function () {}
-    var componentA = { template: 'foo', methods: { f1: f1, f2: function () {} } }
```
Comment (yyx990803) [inline test/unit/specs/util/options_spec.js]: Should use a real constructor here with `var componentA = Vue.extend({ ... })`

Comment (pespantelis) [inline test/unit/specs/util/options_spec.js]: It throws an error:
```js
TypeError: Object prototype may only be an Object or null: undefined
```
I'm working on that, but I cannot face the problem. Any idea? :)

Comment (yyx990803) [inline test/unit/specs/util/options_spec.js]: Can you show the stack trace?

Comment (pespantelis) [inline test/unit/specs/util/options_spec.js]:
```
TypeError: Object prototype may only be an Object or null: undefined
    at create (http://0.0.0.0:8080/specs.js:2111:13)
    at mergeAssets (http://0.0.0.0:8080/specs.js:3440:14)
    at mergeField (http://0.0.0.0:8080/specs.js:3624:21)
    at mergeOptions (http://0.0.0.0:8080/specs.js:3619:8)
    at mergeOptions (http://0.0.0.0:8080/specs.js:3607:56)
    at mergeOptions (http://0.0.0.0:8080/specs.js:3607:109)
    at Object.<anonymous> (http://0.0.0.0:8080/specs.js:25688:16)
    at attemptSync (http://0.0.0.0:8080/lib/jasmine.js:1605:12)
    at QueueRunner.run (http://0.0.0.0:8080/lib/jasmine.js:1593:9)
    at QueueRunner.execute (http://0.0.0.0:8080/lib/jasmine.js:1580:10)
    at Spec.Env.queueRunnerFactory (http://0.0.0.0:8080/lib/jasmine.js:508:35)
    at Spec.execute (http://0.0.0.0:8080/lib/jasmine.js:295:10)
```

**Pull request merged**